osdconfig/interface.go

```diff
@@ -0,0 +1,39 @@
+package osdconfig
+
+import "context"
+
+// A config manager interface allows management of osdconfig parameters
+// It defines setters, getters and callback management functions
+type ConfigManager interface {
+	// Fetch cluster configuration data
+	GetClusterConf() (*ClusterConfig, error)
```

Comment:
All comments for public types and functions need to be start with function/type name, else golint will fail. For e.g

```
// GetClusterConf fetches cluster configuration data...
GetClusterConf() (*ClusterConfig, error)
```

Reply:
sounds good... will do.

Reply:
done